resources/newssite/news-next/src/context/data-context.js

```diff
@@ -0,0 +1,45 @@
+import { createContext, useContext } from "react";
+import { content as contentEn } from "@/data/en/content";
+import { settings as settingsEn } from "@/data/en/dialog";
+import { footer as footerEn } from "@/data/en/footer";
+import * as buttonsEn from "@/data/en/buttons";
+import * as linksEn from "@/data/en/links";
+
+const contentImport = { en: contentEn };
+const settingsImport = { en: settingsEn };
+const footerImport = { en: footerEn };
+const buttonsImport = { en: buttonsEn };
+const linksImport = { en: linksEn };
```

Comment:
I think I'd rather do a structure such as:
```js
const strings = {
  en: {
    content: contentEn,
    settings: settingsEn,
    footer: footerEn,
    buttons: buttonsEn,
    links: linksEn,
  },
};
```
Because it makes it straightforward to check for the existence of the locale later, that is instead of:
```js
const lang = urlParams.get("lang") ?? defaultLanguage;
```
we'd have
```js
const langFromUrl = urlParams.get("lang");
const lang = langFromUrl && langFromUrl in strings ? langFromUrl : defaultLanguage;
```
and then get rid of the condition in the return value:
```js
const value = {
  lang,
  dir,
  ...strings[lang],
};
```

What do you think?

Reply:
I like it - changed.

Reply:
done